lightningd/gossip_control.c

```diff
@@ -510,13 +511,22 @@ static struct command_result *json_dev_query_scids(struct command *cmd,
 	struct node_id *id;
 	struct short_channel_id *scids;
 	size_t i;
+	struct sensitive_command_hook_payload *sensitive_hook;
 
 	if (!param(cmd, buffer, params,
 		   p_req("id", param_node_id, &id),
 		   p_req("scids", param_array, &scidstok),
 		   NULL))
 		return command_param_failed();
 
+	sensitive_hook = tal(cmd, struct sensitive_command_hook_payload);
```

Comment:
If you call the hook in [parse_request](https://github.com/ElementsProject/lightning/blob/4274b9f0afbb919d004be86943f7675aaa1a26cd/lightningd/jsonrpc.c#L586) and don't distinguish between sensitive and non-sensitive you can avoid all this copy-paste in this commit.

Reply:
Yeah, I've almost finished the modifications, indeed this commit was awkward to rebase ^^